examples/extract-auth-token.py

```diff
@@ -4,12 +4,44 @@
 import libflagship.logincache
 import json
 import sys
+import platform
+import os
+import getopt
 
-if len(sys.argv) != 2:
-    print(f"usage: {sys.argv[0]} <path/to/login.json>")
-    exit(1)
+def jankauto():
+    userdir = os.getlogin()
+    useros = platform.system()
 
-data = open(sys.argv[1]).read()
+    darfileloc = f'/Users/{userdir}/Library/Application Support/AnkerMake/AnkerMake_64bit_fp/login.json'
+    winfileloc = os.path.join('C:\\Users\\') + f'{userdir}' + os.path.join('\\AppData\\Local\\Ankermake\\login.json')
+    
+    try:
+        opts, args = getopt.getopt(sys.argv[1:],"haf:", ["help", "auto", "file="])
+    except getopt.GetoptError:
+      print ('extract-auth-token.py -a OR extract-auth-token.py -f <path-to-login.json>')
+      sys.exit(2)
 
-json = json.loads(libflagship.logincache.decrypt(data))
-print(json["data"]["auth_token"])
+    for opt, arg in opts: 
+        if opt == '-h':
+            print ('extract-auth-token.py -a OR extract-auth-token.py -f <path-to-login.json> \nIf spaces are in file location remember to escape them with a backslash')
+            exit(1)
+        elif opt in ("-f", "--file"):
+            data = open(arg).read()
+            jsonj = json.loads(libflagship.logincache.decrypt(data))
+            print(jsonj["data"]["auth_token"])
+            exit(1)
+        elif opt in ("-a", "--auto"):
+            if useros == 'Darwin':
+                data = open(f'{darfileloc}').read()
+                jsonj = json.loads(libflagship.logincache.decrypt(data))
+                print(jsonj["data"]["auth_token"])
+                exit(1)
+            elif useros == 'Windows':
+                data = open(f'{winfileloc}').read()
+                jsonj = json.loads(libflagship.logincache.decrypt(data))
+                print(jsonj["data"]["auth_token"])
+                exit(1)
+        else:
+            assert False, "unhandled option"
+
+jankauto()
```

Comment:
If we're wrapping "everything" in a function, a standardish python idiom is something like:

```python
if __name__ == "__main__":
    exit(main())
```

at the end of the file. That way, the `main()` function can just return whatever exit code is appropriate.